tool/src/main/java/org/wildfly/security/tool/FileSystemEncryptRealmCommand.java

```diff
@@ -0,0 +1,740 @@
+                    descriptor.setSecretKeyAlias("key");
+                    isNew = true;
+                }
+                String csType = PropertiesCredentialStore.NAME;
+                try {
+                    credentialStore = CredentialStore.getInstance(csType);
+                } catch (NoSuchAlgorithmException e) {
+                    // fallback to load all possible providers
+                    credentialStore = CredentialStore.getInstance(csType, getProvidersSupplier(null));
+                }
+                Map<String, String> implProps = new HashMap<>();
+                implProps.put("create", "true");
+                implProps.put("location", descriptor.getCredentialStore());
+                implProps.put("modifiable", Boolean.TRUE.toString());
+                credentialStore.initialize(implProps);
+                try {
+                    credentialStore.retrieve(descriptor.getSecretKeyAlias(), SecretKeyCredential.class).getSecretKey();
+                    System.out.println("Found credential store and alias, using pre-existing key");
+                } catch (Exception e) {
+                    if (!isNew) {
+                        System.out.println("Could not find credential store and secret key alias, exiting");
+                        continue;
+                    }
+                    System.out.println("Credential Store and/or Alias doesn't exists, generating secret key");
+                    SecretKey key = SecretKeyUtil.generateSecretKey(256);
+                    Credential keyCredential = new SecretKeyCredential(key);
+                    credentialStore.store(descriptor.getSecretKeyAlias(), keyCredential);
+                    credentialStore.flush();
+                }
+            } else {
+                throw ElytronToolMessages.msg.cmdFileSystemEncryptionKey();
+            }
+            SecretKey key;
+            try {
+                key = credentialStore.retrieve(descriptor.getSecretKeyAlias(), SecretKeyCredential.class).getSecretKey();
+            } catch (NullPointerException e) {
+                System.out.println("Invalid SecretKey");
+                continue;
+            }
+
+            FileSystemSecurityRealm oldFileSystemRealm = FileSystemSecurityRealm.builder()
+                    .setRoot(Paths.get(descriptor.getInputRealmLocation()))
+                    .setLevels(descriptor.getLevels())
+                    .setHashEncoding(descriptor.getHashEncoding())
+                    .setEncoded(descriptor.getEncoded())
+                    .setProviders(ELYTRON_PASSWORD_PROVIDERS)
+                    .build();
+
+            FileSystemSecurityRealm newFileSystemRealm = FileSystemSecurityRealm.builder()
+                    .setRoot(Paths.get(descriptor.getOutputRealmLocation(), descriptor.getFileSystemRealmName()))
+                    .setSecretKey(key)
+                    .setLevels(descriptor.getLevels())
+                    .setHashEncoding(descriptor.getHashEncoding())
```

Comment:
Does the encrypted realm actually make use of the `hashEncoding`?

Reply:
Good catch it doesn't, I'll remove that